platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/isolated/AbstractIsolatedProjectsToolingApiIntegrationTest.groovy

```diff
@@ -33,4 +33,10 @@ class AbstractIsolatedProjectsToolingApiIntegrationTest extends AbstractIsolated
     GradleExecuter createExecuter() {
         return new ToolingApiBackedGradleExecuter(distribution, temporaryFolder)
     }
+
+    def includeProjects(String... names) {
```

Comment:
❓ 

Did you consider having this in a trait, that then could be reused in more contexts than allowed via implementation inheritance?

Many of the places now calling `createDirs` no longer convey the directories being created are meant to hold subprojects. Hopefully they could use the same shared implementation, or some variant of it.
At a minimum, we could have a `createProjectSubdirs()` that just delegates to `createDirs()`.